x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/data_streams/resolution_table_cell.tsx

```diff
@@ -16,139 +16,167 @@ import {
   EuiFlexItem,
   EuiToolTip,
 } from '@elastic/eui';
-import { DataStreamReindexStatus } from '../../../../../../common/types';
+import {
+  DataStreamMigrationStatus,
+  DataStreamResolutionType,
+} from '../../../../../../common/types';
 import { getDataStreamReindexProgressLabel } from '../../../../lib/utils';
 import { LoadingState } from '../../../types';
-import { useDataStreamReindexContext } from './context';
+import { useDataStreamMigrationContext } from './context';
 
-const i18nTexts = {
-  reindexLoadingStatusText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexLoadingStatusText',
-    {
-      defaultMessage: 'Loading status…',
-    }
-  ),
-  reindexInProgressText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexInProgressText',
-    {
-      defaultMessage: 'Reindexing in progress…',
-    }
-  ),
-  reindexCompleteText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexCompleteText',
-    {
-      defaultMessage: 'Reindex complete',
-    }
-  ),
-  reindexFailedText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexFailedText',
-    {
-      defaultMessage: 'Reindex failed',
-    }
-  ),
-  reindexFetchFailedText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexFetchFailedText',
-    {
-      defaultMessage: 'Reindex status not available',
-    }
-  ),
-  reindexCanceledText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.reindexCanceledText',
-    {
-      defaultMessage: 'Reindex cancelled',
-    }
-  ),
-  resolutionText: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.resolutionLabel',
-    {
-      defaultMessage: 'Reindex',
-    }
-  ),
-  resolutionTooltipLabel: i18n.translate(
-    'xpack.upgradeAssistant.esDeprecations.dataStream.resolutionTooltipLabel',
-    {
-      defaultMessage:
-        'Resolve this issue by reindexing this data stream. This issue can be resolved automatically.',
-    }
-  ),
+const getI18nTexts = (resolutionType?: DataStreamResolutionType) => {
+  return {
+    reindexLoadingStatusText: i18n.translate(
+      'xpack.upgradeAssistant.esDeprecations.dataStream.reindexLoadingStatusText',
+      {
+        defaultMessage: 'Loading status…',
+      }
+    ),
+    reindexInProgressText: i18n.translate(
+      'xpack.upgradeAssistant.esDeprecations.dataStream.reindexInProgressText',
+      {
+        defaultMessage:
+          '{resolutionType, select, reindex {Reindexing} readonly {Marking as readonly} other {Migration}} in progress…',
```

Comment:
```suggestion
          '{resolutionType, select, reindex {Reindexing} read only {Marking as read only} other {Migration}} in progress…',
```

nit we must consistently do `readonly` or `read only`. I think `read only` is probably better.

Reply:
Updated. `readonly` is only used now as the resolution type. the rest is `read only`